### DISCUSSION TITLE
Will it work with a flask app?

### DISCUSSION THREAD

**Discussion by vsisl** [gaogaotiantian/viztracer#383] [Q&A]
Hello!

This looks like a great tool! I am wondering if it could be used to profile a flask app (or more specifically a plotly dash app). Currently, [werkzeug middleware profiler](https://werkzeug.palletsprojects.com/en/2.3.x/middleware/profiler/) together with [snakeviz](https://jiffyclub.github.io/snakeviz/).

I find snakeviz a bit difficult to interpret so I'm curious about alternatives. Would viztracer work?

**Comment by gaogaotiantian** [comment]
Sure, viztracer could work in different ways. I don't think you want to enable the profiler all the time in production - I would strongly recommend against that. But for development, you can run flask with `viztracer -m flask --app hello run` instead of `flask --app hello run`.
